internal/helper/retry.go

```diff
@@ -32,3 +32,29 @@ func Retry(f Retryable, interval time.Duration, attempts int) (int, error) {
 
 	return retries, err
 }
+
+// Retry is a wrapper to retry functions.
+func RetryUntilTimeout(f Retryable, interval time.Duration, timeout time.Duration) (int, error) {
+	var (
+		err         error
+		retry       bool
+		timeElapsed time.Duration
+	)
+
+	timeElapsed = 0 * time.Second
+
+	retries := 0
+
+	for timeElapsed.Seconds() < timeout.Seconds() {
+		retry, err = f()
+		if !retry {
+			break
+		}
```

Comment:
Do we need this logic here? 
Since the retries is getting incremented, will be eventually out of the loop right?

Reply:
Yes, we need to have this logic.
we need it here to exit early if the condition is met